src/qt/bitcoin.cpp

```diff
@@ -135,7 +135,7 @@ BitcoinCore::BitcoinCore(interfaces::Node& node) :
 void BitcoinCore::handleRunawayException(const std::exception *e)
 {
     PrintExceptionContinue(e, "Runaway exception");
-    Q_EMIT runawayException(QString::fromStdString(m_node.getWarnings()));
+    Q_EMIT runawayException(QString::fromStdString(m_node.getWarnings()).replace("<hr />", "\n\n"));
```

Comment:
So we have

https://github.com/bitcoin/bitcoin/blob/88b2652fadf6e004e751d48884ae8d4cf5c452b8/src/warnings.h#L17

https://github.com/bitcoin/bitcoin/blob/88b2652fadf6e004e751d48884ae8d4cf5c452b8/src/warnings.cpp#L45

Would it be better to extract the `warning_separator` variable into a global constant and use it here instead of hardcoding `"<hr />"` in both places?

Even better would be to return a vector of strings instead of multiple strings concatenated to each other with `"<hr />"`, but that would be a much bigger change, I am not suggesting to do it here in this PR.

Reply:
> ... a much bigger change...

That was my concern too :)

Reply:
How about drop `warning_separator` and use `std::endl`?

AFAICT this `<hr />` was introduced in #7579.

Reply:
[Reworked](https://github.com/bitcoin/bitcoin/pull/18898#issuecomment-625027678).